src/components/create-or-edit-question/CreateOrEditQuestion.tsx

```diff
@@ -0,0 +1,192 @@
+
+  const { t } = useTranslation()
+
+  const isMultipleChoice = questionType.value === sortQuestions[0].value
+  const isOpenAnswer = questionType.value === sortQuestions[1].value
+  const isSingleChoice = questionType.value === sortQuestions[2].value
+
+  const setValue = (value: string) => {
+    const questionOption = sortQuestions.find((item) => item.value === value)
+
+    setQuestionType(questionOption ?? sortQuestions[0])
+  }
+
+  const handleQuestion = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuestion(event.target.value)
+  }
+
+  const sortOptions = sortQuestions.map(({ icon, title, value }) => ({
+    title: t(title),
+    value,
+    icon
+  }))
+
+  const handleOptionChange = (
+    event: ChangeEvent<HTMLInputElement>,
+    index: number
+  ) => {
+    const updatedAnswers = [...answers]
+
+    if (isMultipleChoice) {
+      updatedAnswers[index].isCorrect = event.target.checked
+    } else if (isSingleChoice) {
+      updatedAnswers.forEach((answer, i) => {
+        answer.isCorrect = i === index
+      })
+    }
+
+    setAnswers(updatedAnswers)
+  }
+
+  const onChangeInput = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+    index: number
+  ) => {
+    const currentValue = event.target.value
+    setAnswers((prevAnswers) => {
+      const updatedAnswers = [...prevAnswers]
+      updatedAnswers[index] = {
+        ...updatedAnswers[index],
+        text: currentValue
+      }
+      return updatedAnswers
+    })
+  }
+
+  const addNewOneAnswer = (event: MouseEvent<HTMLInputElement>) => {
+    event.preventDefault()
+    setAnswers((prev) => [
+      ...prev,
+      {
+        id: answers.length,
+        text: '',
+        isCorrect: false
+      }
+    ])
+  }
+
+  const deleteRadioButton = (id: number) => {
+    const updatedAnswers = answers.filter((item) => item.id !== id)
+    setAnswers(updatedAnswers)
+  }
+
+  const handleTypeChange = (value: string) => {
+    setAnswers((prevAnswers) =>
+      prevAnswers.map((answer) => ({ ...answer, isCorrect: false }))
+    )
+    setValue(value)
+  }
+
+  const handleAnswer = (event: ChangeEvent<HTMLInputElement>) =>
+    setAnswer(event.target.value)
+
+  const options = answers.map((item) => (
+    <Box key={item.id} sx={styles.answer}>
+      <FormControlLabel
+        checked={item.isCorrect}
+        control={isMultipleChoice ? <Checkbox /> : <Radio />}
+        label={
+          <InputBase
+            fullWidth
+            onChange={(e) => onChangeInput(e, item.id)}
+            placeholder={t('questionPage.writeYourAnswer')}
+            value={item.text}
+          />
+        }
+        onChange={(e) =>
+          handleOptionChange(e as ChangeEvent<HTMLInputElement>, item.id)
+        }
+        sx={styles.inputItem}
+        value={item.id}
+      />
+      <IconButton onClick={() => deleteRadioButton(item.id)}>
+        <CloseIcon data-testid='deleteAnswer' fontSize={SizeEnum.Small} />
```

Comment:
As I know, this icon already has its own data-testid, you can see it in test with screen.debug()

Reply:
Yep, I delete it